src/symbolize/gimli.rs

```diff
@@ -217,8 +220,24 @@ cfg_if::cfg_if! {
                 if !elf.little_endian {
                     return None;
                 }
-                let mut syms = elf.syms.iter().collect::<Vec<_>>();
-                syms.sort_unstable_by_key(|s| s.st_value);
+                let mut syms = elf
+                    .syms
+                    .iter()
+                    .map(|s| (s, false))
+                    .chain(elf.dynsyms.iter().map(|s| (s, true)))
+                    // Only look at function/object symbols. Not entirely sure
+                    // why, but this is what libbacktrace does.
+                    .filter(|(s, _)| {
+                        s.is_function() || s.st_type() == goblin::elf::sym::STT_OBJECT
+                    })
+                    // skip anything that's in an undefined section header,
+                    // again not entirely sure why but this is what libbacktrace
+                    // does.
```

Comment:
If the symbol is undefined then it doesn't have an address in this library. This happens for imported symbols.

Reply:
Oh nice, thanks! I'll update the comment here